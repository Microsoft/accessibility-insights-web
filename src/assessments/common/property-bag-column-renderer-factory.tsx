// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ColumnValueBag } from '../../common/types/property-bag/column-value-bag';
import { IAssessmentInstanceRowData } from '../../DetailsView/components/assessment-instance-table';
import { propertyBagColumnRenderer, PropertyBagColumnRendererConfig } from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static get<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: IAssessmentInstanceRowData<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
