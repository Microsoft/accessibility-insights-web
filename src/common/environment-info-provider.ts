// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { toolName } from '../content/strings/application';
import { ToolData } from './types/store-data/unified-data-interface';

export type EnvironmentInfo = {
    extensionVersion: string;
    browserSpec: string;
    axeCoreVersion: string;
};

export class EnvironmentInfoProvider {
    constructor(
        private extensionVersion: string,
        private browserSpec: string,
        private axeVersion: string,
    ) {}
    public getEnvironmentInfo(): EnvironmentInfo {
        return {
            extensionVersion: this.extensionVersion,
            browserSpec: this.browserSpec,
            axeCoreVersion: this.axeVersion,
        };
    }

    public getToolData(): ToolData {
        return {
            scanEngineProperties: {
                name: 'axe-core',
                version: this.axeVersion,
            },
            applicationProperties: {
                name: toolName,
                version: this.extensionVersion,
            },
        };
    }
}
