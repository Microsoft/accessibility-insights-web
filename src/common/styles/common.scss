// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import 'high-contrast-override.scss';
@import 'colors.scss';
@import 'fonts.scss';

$focus-ring-color: $neutral-100;

@mixin set-focus-ring-styles($property: 'border') {
    #{$property}: 1px solid $focus-ring-color;
    @include ms-high-contrast-override {
        #{$property}: 1px dashed windowtext !important;
    }
}

.ms-Fabric.is-focusVisible .ms-Nav-group .ms-nav-linkButton:focus::after {
    @include set-focus-ring-styles('border');
}

button::after {
    content: '';
    position: absolute;
}

.ms-Button.ms-Button--primary:not(.is-disabled) {
    background: $communication-primary;

    &:hover {
        background: $communication-shade-10;
    }
}

.ms-fontColor-neutralPrimary,
.ms-fontColor-neutralPrimary--hover:hover {
    color: $neutral-100 !important;
}

.ms-Fabric {
    color: $neutral-100;
}

.generic-toggle-component {
    margin-bottom: 4vh;
    .toggle-container {
        margin-bottom: 1vh;
        .toggle-name {
            display: inline-block;
            font-size: 17px;
            font-weight: bold;
        }

        .toggle {
            float: right;
        }
    }
}

.high-contrast-theme {
    .insights-link {
        &:hover {
            text-decoration: underline;
        }
    }
    .is-selected {
        .status-icon {
            color: $high-contrast-icon-color !important;
        }
    }
}

.insights-link {
    // marking important to override office fabric style
    color: $link !important;
    text-decoration: none;

    &:hover {
        color: $link-hover !important;
        text-decoration: underline;
    }
}

.ms-Spinner {
    .ms-Spinner-circle {
        border-top-color: $spinner-text;
    }

    .ms-Spinner-label {
        color: $spinner-text;
    }
}

$normalTitleFontWeight: 300 !default;

$pivotItemLeftBorderWidth: 3px !default;
$pivotItemBorderStyle: solid !default;

$detailsViewHeaderBarHeight: 40px !default;
$detailsViewNavPivotsHeight: 41px !default;
$detailsViewCommandBarHeight: 40px !default;
$detailsViewCommandBarBorderHeight: 1px !default;
$detailsViewLeftNavWidth: 200px !default;
$detailsViewTotalHeaderHeight: (#{$detailsViewCommandBarBorderHeight} + #{$detailsViewCommandBarHeight} + #{$detailsViewHeaderBarHeight});

$fastPassRightPanelMarginRight: 24px !default;
$fastPassRightPanelMarginLeft: 24px !default;
$fastPassRightPanelMarginTop: 24px !default;

@mixin ellipsedText {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
