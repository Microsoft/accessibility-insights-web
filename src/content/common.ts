// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { link } from './link';
import { ContentCreator } from '../views/content/content-page';
import { GuidanceTitle as gt } from '../views/content/guidance-title';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
