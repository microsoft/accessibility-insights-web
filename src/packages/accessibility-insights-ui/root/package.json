{
    "name": "accessibility-insights-ui",
    "version": "0.1.2",
    "description": "Accessibility Insights UI Components",
    "license": "MIT",
    "repository": {
        "type": "git",
        "url": "https://github.com/Microsoft/accessibility-insights-web"
    },
    "dependencies": {
        "axe-core": "4.2.1",
        "classnames": "^2.2.6",
        "lodash": "^4.17.20",
        "luxon": "^1.25.0",
        "office-ui-fabric-react": "7.98.0",
        "react": "^16.14.0",
        "react-dom": "^16.14.0",
        "react-helmet": "^6.1.0",
        "uuid": "^8.3.1"
    }
}
