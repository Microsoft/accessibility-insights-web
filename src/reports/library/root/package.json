{
    "name": "accessibility-insights-report",
    "version": "0.0.3",
    "description": "Accessibility Insights Report",
    "license": "MIT",
    "repository": {
        "type": "git",
        "url": "https://github.com/Microsoft/accessibility-insights-web"
    },
    "dependencies": {
        "applicationinsights-js": "^1.0.20",
        "axe-core": "3.3.2",
        "axios": "^0.19.0",
        "classnames": "^2.2.6",
        "idb-keyval": "^3.2.0",
        "lodash": "^4.17.15",
        "moment": "^2.24.0",
        "office-ui-fabric-react": "6.210.5",
        "q": "1.5.1",
        "react": "^16.12.0",
        "react-copy-to-clipboard": "^5.0.2",
        "react-dom": "^16.12.0",
        "react-helmet": "^5.2.0",
        "react-router-dom": "^5.1.2",
        "ua-parser-js": "^0.7.20",
        "uuid": "^3.3.3",
        "webextension-polyfill-ts": "^0.11.0"
    }
}
