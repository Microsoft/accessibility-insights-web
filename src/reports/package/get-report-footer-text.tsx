// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedFC } from 'common/react/named-fc';
import { ReportFooterTextProps } from 'reports/components/report-sections/report-footer-text';
import { ToolLink } from 'reports/components/report-sections/tool-link';

export const getReportFooterText = (serviceTitle: string) =>
    NamedFC<ReportFooterTextProps>('PackageFooterText', ({ axeCoreVersion, browserSpec }) => <>
        This automated checks result was generated using the {serviceTitle}{' '}
        that helps find some of the most common accessibility issues. The scan was
        performed using axe-core {axeCoreVersion} and {browserSpec}. To more deeply
        investigate your accessibility issues please visit{' '}<ToolLink />.
</>);
