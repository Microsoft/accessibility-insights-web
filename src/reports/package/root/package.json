{
    "name": "temp-accessibility-insights-report",
    "version": "0.1.0",
    "description": "Accessibility Insights Report",
    "license": "MIT",
    "repository": {
        "type": "git",
        "url": "https://github.com/Microsoft/accessibility-insights-web"
    },
    "dependencies": {
        "axe-core": "4.0.1",
        "classnames": "^2.2.6",
        "lodash": "^4.17.20",
        "moment": "^2.27.0",
        "office-ui-fabric-react": "7.98.0",
        "react": "^16.13.1",
        "react-dom": "^16.13.1",
        "react-helmet": "^6.1.0",
        "uuid": "^8.3.0"
    }
}
