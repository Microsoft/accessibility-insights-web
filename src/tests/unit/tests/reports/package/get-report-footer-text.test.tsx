// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { EnvironmentInfo } from 'common/environment-info-provider';
import { shallow } from 'enzyme';
import * as React from 'react';
import { getReportFooterText } from 'reports/package/get-report-footer-text';

describe('getReportFooterText', () => {
    it('renders', () => {
        const environmentInfo: EnvironmentInfo = {
            extensionVersion: '1.2.3',
            browserSpec: 'dummy browser version 1.0.1',
            axeCoreVersion: '4.5.6',
        };

        const Footer = getReportFooterText('THE SERVICE NAME');

        const wrapper = shallow(<Footer {...environmentInfo} />);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
