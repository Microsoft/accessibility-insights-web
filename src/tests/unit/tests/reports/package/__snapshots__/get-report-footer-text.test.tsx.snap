// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getReportFooterText renders 1`] = `
<React.Fragment>
  This automated checks result was generated using the 
  THE SERVICE NAME
   
  that helps find some of the most common accessibility issues. The scan was performed using axe-core 
  4.5.6
   and 
  dummy browser version 1.0.1
  . To more deeply investigate your accessibility issues please visit
   
  <ToolLink />
  .
</React.Fragment>
`;
